include/benchmark/benchmark.h

```diff
@@ -565,17 +565,17 @@ class State {
   int range_y() const { return range(1); }
 
   BENCHMARK_ALWAYS_INLINE
-  size_t iterations() const { return (max_iterations - total_iterations_) + 1; }
+  uint32_t iterations() const { return (max_iterations - total_iterations_) + 1; }
```

Comment:
This change seems unnecessary.  On a 32-bit system it has no effect and on a 64-bit system there is no harm in using `size_t`.  Same comment for all the places that touch iterations counters.

Reply:
With this change, if a benchmark has an overflow bug on 32-bit systems, it will also manifest on a 64-bit system. E.g. `state.SetItemsProcessed(1000000 * state.iterations())` would have identical overflow behavior on both 32-bit and 64-bit systems.

Reply:
And I claim that it's a bad property.

I happen to only ever run benchmarks on 64-bit systems.  Why would you force me to have to deal with 32-bit overflows just for a misconceived notion of uniformity?